drivers/network/tcpip/ip/transport/tcp/tcp.c

```diff
@@ -365,8 +365,12 @@ NTSTATUS TCPConnect
         return Status;
     }
 
-    /* Copy bind address into connection */
-    Connection->AddressFile->Address.Address.IPv4Address = bindaddr.addr;
+    /* Check if we had an unspecified address */
+    if (Connection->AddressFile->Address.Address.IPv4Address != bindaddr.addr)
+    {
+         /* We did, so we need to copy back the address */
+         Connection->AddressFile->Address.Address.IPv4Address = bindaddr.addr;
+    }
```

Comment:
It looks like the assignation could be done in all cases (as was done in the original code).
Indeed, if IPv4Address was different from bindaddr.addr, it would be copied over, and if it's equal, then copying the same thing over should not matter.

Reply:
Done.